InternetHospital.BusinessLogic/Services/AppointmentService.cs

```diff
@@ -54,68 +52,45 @@ public IEnumerable<AppointmentModel> GetMyAppointments(int doctorId)
-            }
-            if ((creationModel.EndTime - creationModel.StartTime).TotalMinutes < MINIMUM_APPOINTMENT_TIME)
-            {
-                return (false, $"Your appointment has to be at least {MINIMUM_APPOINTMENT_TIME} minutes");
-            }
-            if (creationModel.StartTime.Date != creationModel.EndTime.Date)
-            {
-                return (false, "Your appointment can't be bigger then one day");
-            }
-            var isReserved = _context.Appointments.Any(a => a.DoctorId == doctorId
-                                                       && ((creationModel.StartTime >= a.StartTime && creationModel.StartTime < a.EndTime)
-                                                           || (creationModel.EndTime > a.StartTime && creationModel.EndTime <= a.EndTime)));
-            if (isReserved)
-            {
-                return (false, "You already have an appointment for this time");
-            }
-
-            var ifUnfinished = _context.Appointments.Any(a => a.DoctorId == doctorId
-                                                              && (DateTime.Now - a.EndTime).TotalHours > TIME_FOR_FINISHING_APPOINTMENT
-                                                              && (a.StatusId == (int) AppointmentStatuses.RESERVED_STATUS));
-            if (ifUnfinished)
-            {
-                return (false, "You have out of date appointments. Please close it before creating new");
-            }
-
-            return CreateAppointment(creationModel, doctorId) ? (true, "Appointment has been successfully created") : (false, "Error adding an appointment");
+            return CreateAppointment(creationModel, doctorId);
         }
 
         /// <summary>
         /// Get all available appointments of selected doctor 
         /// </summary>
-        /// <param name="doctorId"></param>
+        /// <param name="searchParameters"></param>
         /// <returns>
         /// returns a list of appointments what can be reserved by patient
         /// </returns>
-        public IEnumerable<AvailableAppointmentModel> GetAvailableAppointments(int doctorId)
+        public (IEnumerable<AvailableAppointmentModel> appointments, int quantity)
+            GetAvailableAppointments(AppointmentSearchModel searchParameters)
         {
             var appointments = _context.Appointments
-                .Where(a => (a.DoctorId == doctorId)
-                            && (a.StatusId == (int) AppointmentStatuses.DEFAULT_STATUS)
-                            && (a.StartTime - DateTime.Now).TotalMinutes > MINIMUM_TIME_BEFORE_APPOINTMENT)
-                .Select(a => new AvailableAppointmentModel
-                {
-                    Id = a.Id,
-                    Address = a.Address,
-                    StartTime = a.StartTime,
-                    EndTime = a.EndTime
-                });
-            return appointments.ToList();
+                .AsQueryable()
```

Comment:
It's unnecessary because LINQ to entities working with IQueryable as default interface.